perf/src/sigverify.rs

```diff
@@ -1299,26 +1347,55 @@ mod tests {
     fn test_dedup_same() {
         let tx = test_tx();
 
-        // generate packet vector
         let mut batches =
             to_packet_batches(&std::iter::repeat(tx).take(1024).collect::<Vec<_>>(), 128);
         let packet_count = sigverify::count_packets_in_batches(&batches);
-        let bloom: AtomicBloom<&[u8]> = Bloom::random(1_000_000, 0.0001, 8 << 20).into();
-        let discard = sigverify::dedup_packets(&bloom, &mut batches) as usize;
-        // because dedup uses a threadpool, there maybe up to N threads of txs that go through
-        let n = get_thread_count();
-        assert!(packet_count < discard + n * 2);
+        let filter = Deduper::new(1_000_000, 0);
+        let discard = filter.dedup_packets(&mut batches) as usize;
+        assert_eq!(packet_count, discard + 1);
     }
 
     #[test]
     fn test_dedup_diff() {
-        // generate packet vector
+        let mut filter = Deduper::new(1_000_000, 0);
         let mut batches = to_packet_batches(&(0..1024).map(|_| test_tx()).collect::<Vec<_>>(), 128);
 
-        let bloom: AtomicBloom<&[u8]> = Bloom::random(1_000_000, 0.0001, 8 << 20).into();
-        let discard = sigverify::dedup_packets(&bloom, &mut batches) as usize;
+        let discard = filter.dedup_packets(&mut batches) as usize;
         // because dedup uses a threadpool, there maybe up to N threads of txs that go through
-        let n = get_thread_count();
-        assert!(discard < n * 2);
+        assert_eq!(discard, 0);
+        filter.reset();
+        for i in filter.filter {
+            assert_eq!(i.load(Ordering::Relaxed), 0);
+        }
+    }
+
+    #[test]
+    #[ignore]
+    fn test_dedup_saturated() {
+        let filter = Deduper::new(1_000_000, 0);
+        let mut discard = 0;
+        assert!(!filter.saturated.load(Ordering::Relaxed));
+        for i in 0..1000 {
+            let mut batches =
+                to_packet_batches(&(0..1000).map(|_| test_tx()).collect::<Vec<_>>(), 128);
+            discard += filter.dedup_packets(&mut batches) as usize;
+            println!("{} {}", i, discard);
+            if filter.saturated.load(Ordering::Relaxed) {
+                break;
+            }
+        }
+        assert!(filter.saturated.load(Ordering::Relaxed));
+    }
+    #[test]
+    fn test_dedup_false_positive() {
```

Comment:
nit: need to ignore this right?

Reply:
no, just testing that we don't see a false positive with a small set.  I guess there is some possibility of it happening